test/DisputeManager.test.ts

```diff
@@ -0,0 +1,223 @@
+            await staking.connect(indexer).unstake(indexer.address, etherParse('2'));
+            await staking.connect(indexer).unstake(indexer.address, etherParse('10'));
+            await startNewEra(mockProvider, eraManager);
+            await rewardsDistributor.collectAndDistributeRewards(indexer.address);
+            await rewardsStaking.applyStakeChange(indexer.address, indexer.address);
+
+            expect(await staking.getTotalStakingAmount(indexer.address)).equal(etherParse('1988'));
+            await disputeManager.finalizeDispute(1, 1, etherParse('13'), etherParse('1005'));
+            expect(await staking.getTotalStakingAmount(indexer.address)).equal(etherParse('1987'));
+
+            expect(await token.balanceOf(fisherman.address)).equal(etherParse('1005'));
+
+            expect((await staking.getUnbondingAmounts(indexer.address)).length).equal(0);
+
+            const dispute = await disputeManager.disputes(1);
+            expect(dispute.state).to.equal(1);
+        });
+
+        it('reject dispute should work', async () => {
+            await disputeManager.finalizeDispute(1, 2, 0, etherParse('900'));
+            expect(await staking.getTotalStakingAmount(indexer.address)).equal(etherParse('2000'));
+            expect(await token.balanceOf(fisherman.address)).equal(etherParse('900'));
+
+            const dispute = await disputeManager.disputes(1);
+            expect(dispute.state).to.equal(2);
+        });
+
+        it('cancel dispute should work', async () => {
+            await disputeManager.finalizeDispute(1, 3, 0, etherParse('1000'));
+            expect(await staking.getTotalStakingAmount(indexer.address)).equal(etherParse('2000'));
+            expect(await token.balanceOf(fisherman.address)).equal(etherParse('1000'));
+
+            const dispute = await disputeManager.disputes(1);
+            expect(dispute.state).to.equal(3);
+        });
+
+        it('finalizeDispute with invaild parameter should fail', async () => {
+            await expect(
+                disputeManager.finalizeDispute(1, 1, etherParse('13'), etherParse('2000'))
+            ).to.be.revertedWith('invalid newDeposit');
+
+            await expect(
+                disputeManager.finalizeDispute(1, 1, etherParse('13'), etherParse('900'))
+            ).to.be.revertedWith('invalid newDeposit');
+
+            await expect(
+                disputeManager.finalizeDispute(1, 2, 0, etherParse('1100'))
+            ).to.be.revertedWith('invalid newDeposit');
+
+            await expect(
+                disputeManager.finalizeDispute(1, 3, 0, etherParse('100'))
+            ).to.be.revertedWith('invalid newDeposit');
+        });
+
+        it('indexer cannot widthdraw if on dispute', async () => {
+            await staking.connect(indexer).unstake(indexer.address, etherParse('2'));
+            await startNewEra(mockProvider, eraManager);
```

Comment:
It doesn't necessary to have `startNewEra` , `collectReward` etc, as the check for dispute status is on the first lone of `function withdraw()`, it will revert immediately once the indexer is on dispute